src/sysusers/sysusers.c

```diff
@@ -416,6 +416,8 @@ static int write_temporary_passwd(const char *passwd_path, FILE **tmpfile, char
                 }
                 if (r < 0)
                         return r;
+                if (!r)
```

Comment:
This should be `r != 0`, following the don't-downgrade-ints-to-bool rule.

Reply:
ok but the condition should be `r == 0`

Reply:
Not fixed yet...